src/gadgets/value_shuffle.rs

```diff
@@ -20,22 +20,17 @@ pub fn fill_cs<CS: ConstraintSystem>(
         let x = x[0];
         let y = y[0];
         cs.constrain(y.q - x.q);
-        cs.constrain(y.a - x.a);
-        cs.constrain(y.t - x.t);
+        cs.constrain(y.f - x.f);
         return Ok(());
     }
 
     cs.specify_randomized_constraints(move |cs| {
         let w = cs.challenge_scalar(b"k-value shuffle challenge");
-        let w2 = w * w;
         let mut x_scalars = Vec::with_capacity(k);
         let mut y_scalars = Vec::with_capacity(k);
 
         for i in 0..k {
-            let (x_i_var, y_i_var, _) = cs.multiply(
-                x[i].q + x[i].a * w + x[i].t * w2,
-                y[i].q + y[i].a * w + y[i].t * w2,
-            );
+            let (x_i_var, y_i_var, _) = cs.multiply(x[i].q + x[i].f * w, y[i].q + y[i].f * w);
```

Comment:
Note for later: this is a funny way to allocate two independent variables, but we really should use a more apt API for clarity.